lib/tidewave/mcp/server.ex

```diff
@@ -48,7 +50,10 @@ defmodule Tidewave.MCP.Server do
         true
     end
 
-    for tool <- tools, listable?.(tool) do
+    for tool <- tools,
+        listable?.(tool),
+        is_nil(tool_opts[:include]) or tool.name in tool_opts[:include],
+        is_nil(tool_opts[:exclude]) or tool.name not in tool_opts[:exclude] do
```

Comment:
Above we always set exclude to an empty list but here we check for nil.

Reply:
There seems to be some kind of strangeness with this code:

```elixir
  setup_all do
    start_supervised!({Bandit, plug: Tidewave, port: 9100, startup_log: false},
      shutdown: :brutal_kill
    )

    :ok
  end
```

It doesn't seem to be properly initializing the plug opts, or they appear to be being lost at some point? The observed effect is that if I remove this:

```elixir
is_nil(tool_opts[:exclude]) or tool
```

then for some reason the `Tidewave.McpIntegrationTest`s start failing. I will remove that code and perhaps you might spot where things are going wrong there?